source/Octopus.Client/Repositories/DeploymentSettingsRepository.cs

```diff
@@ -0,0 +1,72 @@
+using Octopus.Client.Model;
+
+namespace Octopus.Client.Repositories
+{
+    public interface IDeploymentSettingsRepository
+    {
+        IDeploymentSettingsRepositoryBeta Beta();
+    }
+
+    class DeploymentSettingsRepository : IDeploymentSettingsRepository
+    {
+        private readonly IDeploymentSettingsRepositoryBeta beta;
+
+        public DeploymentSettingsRepository(IOctopusRepository repository)
+        {
+            beta = new DeploymentSettingsRepositoryBeta(repository);
+        }
+
+        public IDeploymentSettingsRepositoryBeta Beta()
+        {
+            return beta;
+        }
+    }
+
+    public interface IDeploymentSettingsRepositoryBeta
+    {
+        DeploymentSettingsResource Get(ProjectResource project, string gitref = null);
+        DeploymentSettingsResource Modify(ProjectResource project, DeploymentSettingsResource resource, string commitMessage = null);
+    }
+
+    class DeploymentSettingsRepositoryBeta : IDeploymentSettingsRepositoryBeta
+    {
+        private readonly IOctopusRepository repository;
+        private readonly IOctopusClient client;
+
+        public DeploymentSettingsRepositoryBeta(IOctopusRepository repository)
+        {
+            this.repository = repository;
+            client = repository.Client;
+        }
+
+        public DeploymentSettingsResource Get(ProjectResource projectResource, string gitref = null)
```

Comment:
API design choice: should these be based on `ProjectId` or `DeploymentSettingsId`? I defaulted to project.

Reply:
This looks correct. We should follow the existing pattern used by deployment process, since this model aligns nicely with that. We just grab the appropriate link from the project then 👍